src/Pages/ArticlePage.tsx

```diff
@@ -63,40 +63,24 @@ const ArticlePage: React.FC<
 
   useEffect(() => {
     setIsLoadingArticleData(true);
-    if (isLoggedIn) {
-      AXIOS.get(`articles/${slug}`)
-        .then(res => {
-          const article = res.data.article;
-          setArticle(article);
-          return article;
+    AXIOS.get(`articles/${slug}`)
+      .then(res => {
+        const article = res.data.article;
+        setArticle(article);
+        return article;
+      })
+      .then(article =>
+        AXIOS.get(`profiles/${article.author.username}`).then(res => {
```

Comment:
you are not using promise correctly
promise was mainly found to fix the callback hell problem
so for your case, instead of adding another layer of promise callback you can return the promise like this
```
.then(article =>
 return AXIOS.get(`profiles/${article.author.username}`)
})
.then(res => {
 const profile = res.data.profile;
 setProfile(profile);
 setIsLoadingArticleData(false);
})
```